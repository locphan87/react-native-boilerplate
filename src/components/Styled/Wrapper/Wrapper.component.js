// @flow
import { compose, setDisplayName } from 'recompose'
import glamorous from 'glamorous-native'
import { View } from 'react-native'

const styles = (props, { colors }) => ({
  flex: 1,
  backgroundColor: props.bgColor || colors.base,
  alignItems: 'center',
  justifyContent: 'center'
})

export default compose(setDisplayName('StyledWrapper'), glamorous(View))(styles)
