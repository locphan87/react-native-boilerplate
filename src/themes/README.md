# Themes

Application specific themes
* Base Styles
* Fonts
* Metrics
* Colors

