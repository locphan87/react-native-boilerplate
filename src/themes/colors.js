export default {
  base: 'rgba(31, 32, 39, 1)',
  white: 'white',
  yellow: 'yellow',
  gray: '#959299'
}
