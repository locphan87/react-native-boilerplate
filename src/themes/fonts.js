const SIZE = {
  h1: 20,
  h2: 18,
  h3: 16,
  _: 14
}

export default {
  SIZE
}
