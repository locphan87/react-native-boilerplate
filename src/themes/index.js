export { default as COLORS } from './colors'
