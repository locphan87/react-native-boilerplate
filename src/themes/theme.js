import colors from './colors'

const mainTheme = {
  colors
}

export default mainTheme
