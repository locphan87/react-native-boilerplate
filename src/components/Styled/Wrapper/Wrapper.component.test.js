import { testSnapshots } from '../../../utils/test.util'
import { MAIN_THEME } from '../../../themes'

import Wrapper from './Wrapper.component'

const props = {
  theme: MAIN_THEME
}

describe('Styled Wrapper', () => {
  testSnapshots(Wrapper, [
    {
      props,
      description: 'basic render'
    },
    {
      props: {
        ...props,
        bgColor: 'red'
      },
      description: 'custom bg color'
    }
  ])
})
