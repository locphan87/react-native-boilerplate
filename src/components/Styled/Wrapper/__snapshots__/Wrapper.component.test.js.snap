// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Styled Wrapper snapshots basic render 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(31, 32, 39, 1)",
        "flex": 1,
        "justifyContent": "center",
      },
    ]
  }
/>
`;

exports[`Styled Wrapper snapshots custom bg color 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "red",
        "flex": 1,
        "justifyContent": "center",
      },
    ]
  }
/>
`;
